specification/redis/resource-manager/Microsoft.Cache/stable/2023-08-01/redis.json

```diff
@@ -0,0 +1,3383 @@
+      "required": [
+        "name",
+        "type"
+      ]
+    },
+    "UpgradeNotification": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of upgrade notification."
+        },
+        "timestamp": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "Timestamp when upgrade notification occurred."
+        },
+        "upsellNotification": {
+          "readOnly": true,
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Details about this upgrade notification"
+        }
+      },
+      "description": "Properties of upgrade notification."
+    },
+    "NotificationListResponse": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/UpgradeNotification"
+          },
+          "x-ms-identifiers": [
+            "name",
+            "timestamp"
+          ],
+          "description": "List of all notifications."
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Link for next set of notifications."
+        }
+      },
+      "description": "The response of listUpgradeNotifications."
+    },
+    "OperationStatus": {
+      "type": "object",
+      "description": "Asynchronous operation status",
+      "properties": {
+        "properties": {
+          "description": "Additional properties from RP, only when operation is successful",
+          "type": "object",
+          "additionalProperties": {}
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationStatusResult"
+        }
+      ]
+    }
+  },
+  "parameters": {
+    "CacheNameParameter": {
+      "name": "cacheName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the Redis cache.",
+      "x-ms-parameter-location": "method",
+      "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"
+    },
+    "AccessPolicyNameParameter": {
+      "name": "accessPolicyName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the access policy that is being added to the Redis cache.",
+      "x-ms-parameter-location": "method",
+      "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$",
+      "minLength": 3,
+      "maxLength": 63
+    },
+    "AccessPolicyAssignmentNameParameter": {
+      "name": "accessPolicyAssignmentName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the access policy assignment.",
+      "x-ms-parameter-location": "method",
+      "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$",
```

Comment:
Are all the pattern constraints actually applied by the server in existing api-versions? I look at the breaking change warnigns and worry about whether we're adding too many patterns and if some of them might be breaking!

Reply:
@TimLovellSmith the breaking changes are caused due to adding pattern to the cacheNameParameter. The regex is the same as we already define for resource names.

Access policy name and access policy name assignment have only been introduced in API version 2023-05-01-preview before this, and implement the same pattern since then